### PR TITLE
Hyperspectral rotation

### PR DESCRIPTION
**Pull request by HaleySchuhl**
**Describe your changes**
Add new function, based on np.rot90, that is formatted for HSI instances. 

**Type of update**
* New feature or feature enhancement

**Associated issues**

**Additional context**

### CODE REVIEW

**Comment by nfahlgren** [comment]
Be sure to take a look at the DeepSource report, since this PR is based on the main branch the issues it identifies are specific to the PR (unlike those based on 4.x).

File: plantcv/plantcv/hyperspectral/rot90.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Rotate hyperspectral datacubes (in increments of 90 degrees)
```
Comment (jdavidpeery) [inline plantcv/plantcv/hyperspectral/rot90.py:1]: Would add "clockwise", i.e. Rotate hyperspectral datacubes counterclockwise (in ...

File: plantcv/plantcv/hyperspectral/rot90.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    # Extract array data and rotate with numpy function
+    rot_array = np.rot90(m=spectral_data.array_data, k=k)
+    rot_rgb = np.rot90(m=spectral_data.pseudo_rgb, k=k)
+
+    # Dimensions of datacube might change depending on degree of rotation
+    dims = np.shape(rot_array)
+
+    # Create new spectral data object with rotated image data and metadata
+    rot_hsi = Spectral_data(array_data=rot_array, min_wavelength=spectral_data.min_wavelength,
```
[the remark this answers is not in the document]
Comment (HaleySchuhl) [reply plantcv/plantcv/hyperspectral/rot90.py:33]: Not currently but that could be a good idea. Jorge has PR #889 that would introduce the ability to save out modified HSI datacubes, but I see the value in a function to streamline the process of creating a Spectral_data instance. There are some metadata terms that can safely get default values to be used in various places within the hyperspectral sub-package. Would you be willing to write up a GitHub issue regarding this idea? It'll help us remember to talk about it at a pcv dev meeting and facilitate the collaborative note taking :)

File: plantcv/plantcv/hyperspectral/rot90.py (hunk 1 above)
Comment (HaleySchuhl) [reply plantcv/plantcv/hyperspectral/rot90.py:1]: Great point!

**Comment by jdavidpeery** [comment]
Since you're adding a new doc page, does it need to be referenced in the `mkdocs.yml` file?

File: docs/hyperspectral_rot90.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+# Many plants absorb 430nm wavelength light. 
+rot_hsi = pcv.hyperspectral.rot90(spectral_data=spectral_array_obj, k=1)
+
+```
+
+
+**Image after rotating 90 degrees (k=1)**
+
+![Screenshot](img/documentation_images/rotate2/10_rotated_img.jpg)
```
Comment (jdavidpeery) [inline docs/hyperspectral_rot90.md:40]: This image is rotated 10 degrees. It looks like it's the same images that are on the `plantcv.transform.rotate` doc page, where it gets rotated 10 degrees

Comment (HaleySchuhl) [reply docs/hyperspectral_rot90.md:40]: Oh no, good catch. I believe I copy and pasted that doc page to help myself with formatting but forgot to update that line 😅 I'll dig up the actual img I meant to link.

Comment (HaleySchuhl) [reply docs/hyperspectral_rot90.md:40]: Jk, looks like I never created an example image. We can just delete the example images from this doc page. None of our other HSI docs have example images either.

File: plantcv/plantcv/hyperspectral/rot90.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (HaleySchuhl) [reply plantcv/plantcv/hyperspectral/rot90.py:33]: See issue #896

File: docs/hyperspectral_rot90.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+```python
+
+from plantcv import plantcv as pcv
+
+# Set global debug behavior to None (default), "print" (to file), 
+# or "plot" (Jupyter Notebooks or X11)
+pcv.params.debug = "plot"
+
+# Many plants absorb 430nm wavelength light.
```
Comment (jdavidpeery) [inline docs/hyperspectral_rot90.md:29]: Guess I missed this the first time, is this comment also a leftover from a copy-pasted page?

**Review by jdavidpeery — APPROVED** [review]
Looks good! Already added and resolved all my comments

**Pull request merged**